include/bluetooth/audio/hap.h

```diff
@@ -0,0 +1,275 @@
+	 * not be set, but if @kconfig{BT_HAP_HARC_AUTO_READ_ACTIVE_PRESET}
+	 * option is enabled, the callback will be called again on preset read
+	 * complete.
+	 *
+	 * @param hap Profile instance object.
+	 * @param preset Preset record object.
+	 */
+	void (*active_preset_changed)(struct bt_hap *hap,
+				      const struct bt_has_preset *preset);
+
+	/**
+	 * @brief A cached preset list has been updated.
+	 *
+	 * This callback notifies the application when the list of preset
+	 * records changed i.e. record has been added/removed, changed name or
+	 * become available.
+	 * The user can call @ref bt_hap_preset_foreach to obtain the current
+	 * presets.
+	 *
+	 * @param hap Profile instance object.
+	 */
+	void (*preset_cache_updated)(struct bt_hap *hap);
+};
+
+/**
+ * @def BT_HAP_CB_DEFINE
+ *
+ * @brief Register a callback structure for Hearing Access Profile events.
+ *
+ * @param _name Name of callback structure.
+ */
+#define BT_HAP_CB_DEFINE(_name)							\
+	static const STRUCT_SECTION_ITERABLE(bt_hap_cb,				\
+					     _CONCAT(bt_hap_cb_, _name))
+
+/**
+ * @brief Create profile instance to remote Hearing Access Service
+ *
+ * This will start a GATT discovery to setup handles and subscriptions.
+ * This shall be called once before any other actions can be executed for the
+ * peer device, and the @ref bt_hap_cb connected callback will notify
+ * when it is possible to start remote operations.
+ * The profile instance is removed when the bond is removed or when @ref
+ * bt_hap_delete is called.
+ * 
+ * @param conn The connection to initialize the profile for.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int bt_hap_new(struct bt_conn *conn);
+
+/**
+ * @brief Delete profile instance
+ *
+ * This will remove stored profile instance of remote Hearing Aid Service.
+ *
+ * @param hap Profile instance object.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int bt_hap_delete(struct bt_hap *hap);
+
+/**
+ * @brief Iterate through all existing profile instances.
+ *
+ * Iterate profile instances of remote Hearing Access Services.
+ *
+ * @param func Function to call for each profile instance.
+ * @param user_data Data to pass to the callback function.
+ */
+void bt_hap_foreach(bool (*func)(const struct bt_hap *hap, void *user_data),
+		    void *user_data);
+
+/**
+ * @brief Get profile instance to local Hearing Access Service.
+ *
+ * Get the profile instance connection to local Hearing Access Service.
+ *
+ * @return Profile instance object or NULL if there is no service.
+ */
+struct bt_hap *bt_hap_local_get(void);
+
+/**
+ * @brief Get the Bluetooth connection object of the profile instance.
+ *
+ * Get the Bluetooth connection object of a HAP instance.
+ *
+ * @param hap Profile instance object.
+ *
+ * @return Bluetooth connection object or NULL.
+ */
+struct bt_conn *bt_hap_conn_get(const struct bt_hap *hap);
+
+/** 
+ * @typedef bt_hap_complete_func_t
```

Comment:
```suggestion
 * @typedef bt_hap_complete_func
```

Reply:
Actually I disagree, because I saw the the suffix *_t is used for function typedefs in the code

Reply:
It appears that you might be correct. I can't find anything specific for this in the code style, nor is is common to omit `_t` in the public headers. 

I mentioned it, because I've been told by reviewers before to omit it, but that was a long time ago and I can't remember which reviewer it was, nor the reasoning. 

Leave the `_t` for now, and I suppose we might want to add it to the other LE Audio implementations which do omit it.